compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -365,12 +365,12 @@ OMR::X86::CodeGenerator::initialize(TR::Compilation *comp)
    self()->setSupportsArrayCmp();
    self()->setSupportsArrayCopy();
 
-   if (comp->getOption(TR_EnableX86AdvancedMemorySet))
-      self()->setSupportsArraySet();
-
    static char *disableArraySet = feGetEnv("TR_disableArraySet");
```

Comment:
It is historical code. I'm not sure why it was designed that way but I guess the env var was purely for debugging when it was first developed and meant to be deleted later. Z has neither env var or option; P has SupportsArraySet off by default. Some optimizer has its own way to turn arrayset related optimization on/off, for example TR_DisableSeparateInitFromAlloc.